students/cbrown/Lesson 3/mailroom.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+#Mailroom script part one. Using list data structures
+import sys
+
+donors = [("Bill Gates",[539000,235642]),
+          ("Jeff Bezos",[108356,204295,897345]),
+          ("Satya Nadella",[236000,305352]),
+          ("Mark Zuckerberg",[153956.35]),
+          ("Mark Cuban",[459035,369.50,570.89])]
+
+prompt = "\n".join(("Please Select from Items Below:",
+                    "1 - Send A Thank You",
+                    "2 - Create A Report",
+                    "3 - Quit",
+                    " "))
+
+def send_thanks():
+    print('\n',"For A Complete Donor List Type 'List'\n ")
+    donor_list = []
+
+    #create a unique list of donor names
+    for donor in donors:
+        name = donor[0]
+        donor_list.append(name)
+
+    while True:
+        donor_name = input("What donor(s) are you looking for?\n ").title()
+        if donor_name == 'List':
+            print('\n',"Here is A Complete List of Donor Names\n ")
+            print(donor_list)
+        else:
+            break
+
+    #prompt for donation amount
+    amount = float(input("How Much Did This Person Donate?\n "))
+
+    #Donor name found in list
+    if donor_name in donor_list:
+        for name in donors:
+            if name[0] == donor_name:
+                name[1].append(amount)
+                print(donors)
+
+            else:
+                continue
+
+    #Donor name not found in list
+    else:
+        tup = (donor_name,amount)
+        donors.append(tup)
```

Comment:
this loop is a bit inefficient - the `donor_name in donor_list` will search the whole list, then `for name in donors` will do the search again.

You can just have a single loop to accomplish the same.

```
for donor in donors:
    if name[0] == donor_name:
        donor[1].append(amount)
        break
else:
    # not found
    donors.append((donor_name, amount))
```